
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Shasta assembler user guides and documentation">
      
      
      
        <link rel="canonical" href="https://chanzuckerberg.github.io/shasta-docs/faq/">
      
      <link rel="icon" href="../img/favicon.ico">
      <meta name="generator" content="mkdocs-1.3.0, mkdocs-material-8.3.9">
    
    
      
        <title>FAQ - Shasta User Guides</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.1d29e8d0.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
      
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#faq" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Shasta User Guides" class="md-header__button md-logo" aria-label="Shasta User Guides" data-md-component="logo">
      
  <img src="../assets/ShastaIconColor.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Shasta User Guides
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              FAQ
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Shasta User Guides" class="md-nav__button md-logo" aria-label="Shasta User Guides" data-md-component="logo">
      
  <img src="../assets/ShastaIconColor.svg" alt="logo">

    </a>
    Shasta User Guides
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../public_assemblies/" class="md-nav__link">
        Public Assemblies
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          FAQ
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        FAQ
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#do-i-need-to-preprocess-my-reads" class="md-nav__link">
    Do I need to preprocess my reads?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#does-it-matter-what-version-of-guppy-i-used-to-basecall-my-reads" class="md-nav__link">
    Does it matter what version of Guppy I used to basecall my reads?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#can-i-run-shasta-using-docker" class="md-nav__link">
    Can I run Shasta using Docker?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-is-the-minimum-read-coverage-required-for-standard-ont-reads" class="md-nav__link">
    What is the minimum read coverage required for standard ONT reads?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-is-the-minimum-read-coverage-required-for-ultra-long-ont-reads" class="md-nav__link">
    What is the minimum read coverage required for ultra-long ONT reads?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-much-memory-do-i-need-for-my-genome" class="md-nav__link">
    How much memory do I need for my genome?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-do-i-choose-a-config" class="md-nav__link">
    How do I choose a config?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#are-diploid-phased-genome-assemblies-supported" class="md-nav__link">
    Are diploid phased genome assemblies supported?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#my-assembly-sizecontiguity-is-not-what-i-expected-what-parameters-can-i-tweak" class="md-nav__link">
    My assembly size/contiguity is not what I expected; what parameters can I tweak?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-polisher-do-you-recommend" class="md-nav__link">
    What polisher do you recommend?
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="https://chanzuckerberg.github.io/shasta/" class="md-nav__link">
        Shasta Documentation
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../about/" class="md-nav__link">
        About
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#do-i-need-to-preprocess-my-reads" class="md-nav__link">
    Do I need to preprocess my reads?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#does-it-matter-what-version-of-guppy-i-used-to-basecall-my-reads" class="md-nav__link">
    Does it matter what version of Guppy I used to basecall my reads?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#can-i-run-shasta-using-docker" class="md-nav__link">
    Can I run Shasta using Docker?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-is-the-minimum-read-coverage-required-for-standard-ont-reads" class="md-nav__link">
    What is the minimum read coverage required for standard ONT reads?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-is-the-minimum-read-coverage-required-for-ultra-long-ont-reads" class="md-nav__link">
    What is the minimum read coverage required for ultra-long ONT reads?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-much-memory-do-i-need-for-my-genome" class="md-nav__link">
    How much memory do I need for my genome?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-do-i-choose-a-config" class="md-nav__link">
    How do I choose a config?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#are-diploid-phased-genome-assemblies-supported" class="md-nav__link">
    Are diploid phased genome assemblies supported?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#my-assembly-sizecontiguity-is-not-what-i-expected-what-parameters-can-i-tweak" class="md-nav__link">
    My assembly size/contiguity is not what I expected; what parameters can I tweak?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-polisher-do-you-recommend" class="md-nav__link">
    What polisher do you recommend?
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                


<h1 id="faq">FAQ<a class="headerlink" href="#faq" title="Permanent link">&para;</a></h1>
<h2 id="do-i-need-to-preprocess-my-reads">Do I need to preprocess my reads?<a class="headerlink" href="#do-i-need-to-preprocess-my-reads" title="Permanent link">&para;</a></h2>
<p>No. Shasta accepts unprocessed reads as input in .fasta or .fastq format. Compressed files are not supported, so make sure you decompress them before running Shasta.</p>
<p>Any reads shorter than the Reads.minReadLength (default 10000) containing bases with repeat counts &gt;255 are automatically discarded. You can adjust the minimum read length in the assembly using the Reads.minReadLength.</p>
<h2 id="does-it-matter-what-version-of-guppy-i-used-to-basecall-my-reads">Does it matter what version of Guppy I used to basecall my reads?<a class="headerlink" href="#does-it-matter-what-version-of-guppy-i-used-to-basecall-my-reads" title="Permanent link">&para;</a></h2>
<p>Yes. For optimal results, please basecall your ONT reads with the latest version of Guppy 5 or 6 using the super accuracy mode. If your reads are Guppy 4 or earlier, it is worth re-basecalling them before running an assembly.</p>
<h2 id="can-i-run-shasta-using-docker">Can I run Shasta using Docker?<a class="headerlink" href="#can-i-run-shasta-using-docker" title="Permanent link">&para;</a></h2>
<p>Yes! Documentation can be found in the <a href="https://chanzuckerberg.github.io/shasta/Docker.html" target="_blank">Shasta man pages</a></p>
<h2 id="what-is-the-minimum-read-coverage-required-for-standard-ont-reads">What is the minimum read coverage required for standard ONT reads?<a class="headerlink" href="#what-is-the-minimum-read-coverage-required-for-standard-ont-reads" title="Permanent link">&para;</a></h2>
<p>For standard ONT reads, Shasta performs well with at least 30x coverage of reads ≥10Kb. Optimal coverage is 50-80x of reads ≥10Kb.</p>
<p>You can adjust the coverage by specifying the desired value for total coverage, in bases with the command --Reads.desiredCoverage. For example, to adjust the desired coverage for a 1 Gb genome to 60x, use --Reads.desiredCoverage 60G.</p>
<h2 id="what-is-the-minimum-read-coverage-required-for-ultra-long-ont-reads">What is the minimum read coverage required for ultra-long ONT reads?<a class="headerlink" href="#what-is-the-minimum-read-coverage-required-for-ultra-long-ont-reads" title="Permanent link">&para;</a></h2>
<p>The optimal coverage for ultra-long (UL) reads is 60-80x of reads N50 ≥ 50Kb.</p>
<h2 id="how-much-memory-do-i-need-for-my-genome">How much memory do I need for my genome?<a class="headerlink" href="#how-much-memory-do-i-need-for-my-genome" title="Permanent link">&para;</a></h2>
<p>Shasta works on a single large machine and operates in memory.</p>
<p>Memory requirements are roughly proportional to genome size and coverage and are around 4-6 bytes per input base. For a human-size genome (~3 Gb) at 60x, this works out to ~1 TB of memory.</p>
<p>Machines with 1-24 TB of memory are available at reasonable prices on cloud computing platforms (e.g., AWS EC2 instances). For all assemblies requiring up to 1 TB, you should use ARM x2gd instance types. These instances are available at ~$3/hour as on-demand instances. They complete a human assembly at coverage 60x in ~5 hours, at the cost of ~$15 per assembly.</p>
<p>See our <a href="https://chanzuckerberg.github.io/shasta/Running.html#LowMemory" target="_blank">documentation</a> for more detailed information on memory requirements, memory modes and running Shasta with less than optimal memory (i.e,. for very small genomes).</p>
<h2 id="how-do-i-choose-a-config">How do I choose a config?<a class="headerlink" href="#how-do-i-choose-a-config" title="Permanent link">&para;</a></h2>
<p>As a starting point, please use the latest assembly configuration corresponding to the length of reads you have and the type of assembly you want to run. For example, if you have standard ONT reads and want to run a haploid assembly, please use Nanopore-May2022. If you're going to run a haploid assembly with ultra-long ONT reads, please use Nanopore-UL-May2022. For more detail please see the <a href="https://chanzuckerberg.github.io/shasta/Configurations.html" target="_blank">documentation</a>.</p>
<h2 id="are-diploid-phased-genome-assemblies-supported">Are diploid phased genome assemblies supported?<a class="headerlink" href="#are-diploid-phased-genome-assemblies-supported" title="Permanent link">&para;</a></h2>
<p>Yes. However, this algorithm is being actively developed and improved. Please run phased assemblies using the appropriate config (i.e., Nanopore-Phased-May2022 or Nanopore-UL-Phased-May2022). For more information, see the docs.</p>
<h2 id="my-assembly-sizecontiguity-is-not-what-i-expected-what-parameters-can-i-tweak">My assembly size/contiguity is not what I expected; what parameters can I tweak?<a class="headerlink" href="#my-assembly-sizecontiguity-is-not-what-i-expected-what-parameters-can-i-tweak" title="Permanent link">&para;</a></h2>
<p>If there is a reference for your species or you have an estimate of the genome size from flow cytometry or karyotyping, then the size of your Shasta assembly should be close to the estimated size. Although, we usually find that Shasta assemblies are slightly smaller than the estimated size. You can try tweaking the read length cutoff --Reads.minReadLength. The default is 10000 for standard ONT reads. If you have low coverage (i.e., less than 60x), you could lower the --Reads.minReadLength (in 2,500-5,000 intervals) increases the amount of coverage used in the assembly and potentially increases the assembly size. If you have reads with 60-80x coverage, you can try increasing the --Reads.minReadLength (in 2,500-5,000 intervals) to remove smaller reads from the assembly and potentially increase the contiguity.</p>
<p>If the above recipe does produce the desired result we recommend you, 1) increase the number of MinHash iterations, for example to 100, 2) set the read graph creation method to 0 (no adaptivity in selecting alignment criteria), 3. experiment with alignment criteria, starting with the values chosen in the out of the box assembly (they are in the assembly log). Generally one needs to increase both minAlignedMarkerCount and minAlignedFraction.</p>
<h2 id="what-polisher-do-you-recommend">What polisher do you recommend?<a class="headerlink" href="#what-polisher-do-you-recommend" title="Permanent link">&para;</a></h2>
<p>For human genomes, we recommend <a href="https://github.com/kishwarshafin/pepper/tree/refs/tags/r0.8" target="_blank">Pepper-Margin-DeepVariant</a>. For non-humans, we have had good results using one round of the <a href="https://github.com/fenderglass/Flye/blob/flye/docs/USAGE.md#running-only-flye-polisher" target="_blank">Flye polisher</a>, but results may vary.</p>

              
            </article>
            
          </div>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../public_assemblies/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Public Assemblies" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Public Assemblies
            </div>
          </div>
        </a>
      
      
        
        <a href="../about/" class="md-footer__link md-footer__link--next" aria-label="Next: About" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              About
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2022 Chan Zuckerberg Initiative Foundation
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.6c7ad80a.min.js"></script>
      
    
  </body>
</html>